tests/ttnn/unit_tests/operations/test_distributed_layernorm_sharded.py

```diff
@@ -0,0 +1,472 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+import ttnn
+import torch
+import pytest
+from loguru import logger
+
+from models.utility_functions import (
+    skip_for_wormhole_b0,
+    comp_allclose_and_pcc,
+    comp_pcc,
+    comp_allclose,
+)
+
+from models.utility_functions import tt2torch_tensor, get_devices_for_t3000, skip_for_grayskull
+
+
+def rms_norm(x, gamma, eps):
+    return x * torch.rsqrt(x.pow(2).mean(-1, keepdim=True) + eps) * gamma
+
+
+def layer_norm(x, gamma, eps):
+    return (x - x.mean(-1, keepdim=True)) * torch.rsqrt(x.var(-1, keepdim=True) + eps) * gamma
+
+
+@skip_for_grayskull()
+@pytest.mark.parametrize("is_rmsnorm", [True, False])
+@pytest.mark.parametrize("seed", [0, 2000, 50000])  # Test across 5 different seeds
+@pytest.mark.parametrize("eps", [1e-6])
+@pytest.mark.parametrize("min_pcc", [0.9997])
+@pytest.mark.parametrize("max_atol", [0.38])
+@pytest.mark.parametrize("input_width", [1024, 2048])
+@pytest.mark.parametrize("input_df", [ttnn.bfloat8_b, ttnn.bfloat16])
+@pytest.mark.parametrize("weights_df", [ttnn.bfloat8_b, ttnn.bfloat16])
+@pytest.mark.parametrize(("mean", "std"), ([0, 1], [10.0, 11.0], [100.0, 110.0]))
+def test_sharded_layernorm(
```

Comment:
This tests sharded layernorm. Shouldn't this file only test distributed sharded?

Reply:
Yeah, I was actually using this test to verify generality and correctness. Now I can remove this test

Reply:
@cglagovichTT, I heavily refactored test script to make it more modular and should address most of your comments.
1. Removed non-distributed LN test
2. Removed irrelevant perf_test
3. Added e2e test simulating distributed layernorm on single device
4. Added PCC and ATOL bounds